nextest-runner/src/config/global_timeout.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) The nextest Contributors
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use serde::{Deserialize, de::IntoDeserializer};
+use std::{fmt, time::Duration};
+
+/// Type for the global-timeout config key.
+#[derive(Clone, Copy, Debug, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "kebab-case")]
+pub struct GlobalTimeout {
+    #[serde(with = "humantime_serde")]
+    pub(crate) period: Duration,
+}
```

Comment:
Is there a reason to accept a table with `period` here? That makes more sense for `slow-timeout` which takes a `terminate-after`.

I think `global-timeout` can just accept a scalar `Duration` given that there isn't a great use case for marking an entire test run slow, just one that times out.